contracts/test/PDPService.t.sol

```diff
@@ -70,6 +78,32 @@ contract PDPServiceProofSetCreateDeleteTest is Test {
     }
 }
 
+contract PDPServiceProofSetMutateTest is Test {
+    PDPService pdpService;
+
+    function setUp() public {
+        pdpService = new PDPService(2);
+    }
+
+    function testAddRoot() public {
+        uint256 setId = pdpService.createProofSet();
+        PDPService.RootData[] memory roots = new PDPService.RootData[](1);
+        roots[0] = PDPService.RootData(Cids.Cid(abi.encodePacked("test")), 64);
+        uint256 rootId = pdpService.addRoots(setId, roots);
+        
+        uint256 leafCount = roots[0].rawSize / 32;
+        assertEq(pdpService.getProofSetLeafCount(setId), leafCount);
+        assertEq(pdpService.getNextChallengeEpoch(setId), block.number + 2);
+
+        assertTrue(pdpService.rootLive(setId, rootId));
+        assertEq(pdpService.getRootCid(setId, rootId).data, roots[0].root.data);
+        assertEq(pdpService.getRootLeafCount(setId, rootId), leafCount);
+    }
+
+    // TODO: test bad inputs to addRoot
+    // TODO: test removing roots, good and bad inputs
```

Comment:
There already exist tests removing roots though I agree we don't have as much testing as I'd like I thought that this fell more under fuzzing than unit tests. What else are you specifically looking for here?

Reply:
(https://github.com/FILCAT/pdp/pull/38/files#diff-5dd05f8e5c9c7b2856abd89d1eaeeafae1f35807caf7369c7f1748df051ce89bR220)

Reply:
I think I get it, you'd like a more thorough check of all state variable changed by adding / removing.  Can do